### ISSUE TITLE
Optimize serialization

### ISSUE THREAD

**Issue by nkev** [pubsubsql/client#1]
In addition to [JSON](https://github.com/pubsubsql/client/blob/master/client.go#L113), implementing [gogoprotobuf](https://github.com/gogo/protobuf) for marshaling/unmarshalling will be [10-15 times faster](https://github.com/alecthomas/go_serialization_benchmarks). The easiest way to do this would be to use Googles [gRPC](http://www.grpc.io/) between PubSubSQL and the various client implementations. gRPC is really starting to take off (even before being released!).

**Comment by pssadmin** [comment]
Agreed! JSON needs to be ditched along with optimizing the communication frame altogether. Except for the NodeJS community, no one benefits from JSON. This also means we need to do the some for other clients (nodejs, python, java) so it's a decent amount of work. Therefore, we should start by supporting both formats and ditch JSON when ready on all fronts.
